New guide: mask shorthand

estelle opened the pull request:
i am working on a much deeper guide about all the properties, so that link is red in this PR, but since this is done and that is taking a while, PRing this now.

estelle committed:
New guide: mask shorthand

estelle committed:
title

Railore committed:
Wrong reference to upgrade-insecure-requests header (#39740)

* Wrong reference to upgrade-insecure-requests header

The link points to frame-ancestors header instead.

* Update files/en-us/web/security/practical_implementation_guides/csp/index.md

---------

Co-authored-by: wbamberg <will@bootbonnet.ca>

wbamberg committed:
Update attacks nav (#39736)

* Delete types of attack and CSRF prevention, redirect to new pages

* Rewrite CSRF glossary page

* Remove old clickjacking guide, redirect to new one

* try to fix wikihistory:

* Fix some broken links

* Link to OWASP for session fixation

* Apply suggestions from code review

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

---------

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

twittwer committed:
Add constraint for initial value in @property rule (#38321)

* Add constraint for initial value in @property rule

Adds the contraint of computationally independence for the initial value of @property rules.
Spec-Reference: https://www.w3.org/TR/css-properties-values-api-1/#initial-value-descriptor

* Apply suggestions from code review

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

* Apply suggestions from code review

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>
Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

Josh-Cena committed:
Fix invalid CSS syntax (#39738)

* Fix invalid CSS syntax

* Update files/en-us/web/css/transform-function/rotate3d/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

* Update files/en-us/web/css/transform-origin/index.md

---------

Co-authored-by: Brian Smith <brian@smith.berlin>

Josh-Cena committed:
Extract inline script/style into separate CSS/HTML blocks (#39744)

Josh-Cena committed:
Fix spelling, remove unused words (#39749)

bsmth committed:
chore: Change ints to 'number' for ProgressEvent properties (#39574)

* chore: Change ints to doubles for ProgressEvent properties

* Apply suggestions from code review

* Apply suggestions from code review

* Update files/en-us/web/api/progressevent/loaded/index.md

* feat: Clarify when progressevents use decimals

* Apply suggestions from code review

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

* feat: Clarify when progressevents use decimals

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

kannanwisen committed:
fit-content: explain browser computation (#39730)

* Update index.md

* Update files/en-us/web/css/fit-content/index.md

---------

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>
Co-authored-by: Estelle Weyl <estelle@weyl.org>

Josh-Cena committed:
Internalize closure example jsfiddle (#39728)

samuel871211 committed:
doc: add explanation for multiple `Transfer-Encoding` values (#39496)

* doc: add explanation for multiple `Transfer-Encoding` values

* Clarify use

* Update files/en-us/web/http/reference/headers/transfer-encoding/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update files/en-us/web/http/reference/headers/transfer-encoding/index.md

* Update files/en-us/web/http/reference/headers/transfer-encoding/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

* Update files/en-us/web/http/reference/headers/transfer-encoding/index.md

* Update files/en-us/web/http/reference/headers/transfer-encoding/index.md

* Update files/en-us/web/http/reference/headers/transfer-encoding/index.md

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

---------

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Brian Smith <brian@smith.berlin>

mdn-bot committed:
fix: auto-cleanup by bot (#39752)

* chore: auto-fix Markdownlint, Prettier, and front-matter issues

* Fix one more

* Add more

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

yin1999 committed:
fix typo (#39753)

Commit:
fix: correct casing in title to match other element pages (#39756)

SimonSiefke committed:
Update document.adoptedStyleSheets to use newer push method (#39711)

Update index.md

dletorey committed:
corrected spelling and updated example to show result on page (#39750)

* corrected spelling and updated example to show result on page

* Update files/en-us/web/api/highlight/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* updated heading structure

* updated html in examplle to reset the ranges

* added nolint to stop linter altering the positions of the ranges

* Update files/en-us/web/api/highlight/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

* added the s back to examples

* Update files/en-us/web/api/highlight/index.md

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Joshua Chen <sidachen2003@gmail.com>
Co-authored-by: Brian Smith <brian@smith.berlin>

karelbilek committed:
Replace deprecated substr with slice (#39771)

string.substr() is deprecated; string.slice() is not

dletorey committed:
39619 css highlight api ff release (#39764)

* added Firefox release note for CSS Highlight API

* Update files/en-us/mozilla/firefox/releases/140/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update files/en-us/mozilla/firefox/releases/140/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

rubywwwilde committed:
Add missing argument for the fromOffsetsToSelection function (#39773)

I tried to use the provided code, but it throws an error. According to source code, editor element should be passed as the third  argument: https://github.com/mdn/dom-examples/blob/3f316b6be2e29596b1dfb2e82e358167cef6c372/edit-context/html-editor/editor.js#L106

Josh-Cena committed:
Add allow-forms to example (#39760)

mfuji09 committed:
Add Japanese translation for the accessibility sidebar (#39770)

bsmth committed:
chore(fx): Temporal no longer experimental in Fx (#39776)

mfuji09 committed:
Add and update Japanese translations for the CSS sidebar (#39775)

* Add and update Japanese translations for the CSS sidebar

* Update cssref.yaml

mfuji09 committed:
Add and update Japanese translations for learn sidebar (#39774)

Josh-Cena committed:
Remove inline script/style (API A–F) (#39757)

* Remove inline script/style

* Update files/en-us/web/privacy/guides/firefox_tracking_protection/index.md

Co-authored-by: Brian Smith <brian@smith.berlin>

* Two example

---------

Co-authored-by: Brian Smith <brian@smith.berlin>

dipikabh commented on files/en-us/web/css/css_masking/multiple_masks/index.md:
```diff
@@ -0,0 +1,181 @@
+
+`<image> <position> / <size> <repeat> <origin> <clip> <composite> <mode>`
+
+For the most part, the order of the properties is flexible, but there are a few quirks and exceptions.
+
+The `mask-origin` and `mask-clip` values are similar. Both accept [`<geometry-box>`](/en-US/docs/Web/CSS/box-edge#geometry-box) values, with `mask-clip` also supporting `no-clip`. If one `<geometry-box>` value and the `no-clip` keyword are present then the `<geometry-box>` sets `mask-origin` value and `mask-clip` is set to `no-clip`. In this case the order doesn't matter. If only one `<geometry-b ...
```
```suggestion
For the most part, the order of the properties is flexible, but there are a few quirks and exceptions:

- The `mask-origin` and `mask-clip` values are similar. Both accept [`<geometry-box>`](/en-US/docs/Web/CSS/box-edge#geometry-box) values, with `mask-clip` also supporting `no-clip`.
  - If one `<geometry-box>` value and the `no-clip` keyword are present, then the `<geometry-box>` sets `mask-origin` value and `mask-clip` is set to `no-clip`. In this case, the order doesn't matter.
  - If only one `<geometry-box>` value is present (and there is no `no-clip`), but the `mask-origin` and `mask-clip` components are set to that value. As there is only one value, no order matters here either.
- The order matters only if two `<geometry-box>` values are present; in this case, the first `<geometry-box>` sets the `mask-origin` component and the second `<geometry-box>` value is the `mask-clip` component.
```

dipikabh reviewed (changes requested):
Thanks for adding this guide, @estelle!

I've done an initial readthrough and have a few requests before I send a more detailed review:
- Could you do a quick proofread to fix some obvious typos, like "Each mask layers consists of...", "...as it's value", "...creates a mask layers", "impoorted" in a link text, missing comma after "In this guide". "The minimum required..." should probably be "The minimum requirement".
- The first sentence is quite long and could be split up for better readability and clarity.
- I really like the masquerade ball analogy, but I wonder if "masquerade balls" is understood universally. Do you think we can use another example?
- The page is quite long. Can we consider trimming (removing) parts that are covered in other guides; for example:
  - Could we remove the explanation of what masks are and the analogy to the intro guide and link to it from here?
  - The first four paragraphs can be compressed to briefly set the context and then jump into the main topic of declaring multiple masks. I feel we're not getting to it quickly enough.
- The section headings can be more descriptive; for example, "Overview of the `mask-image` property" and "Declaring mask images using the `mask` shorthand".
- The "Layer syntax" section feels like the core of this guide. Should that be at the `##` level? And possibly renamed to "Understanding the mask layer syntax"? 
- Also wondering about structure: should we introduce the layer syntax before covering all the individual component properties, and then explain the `mask` constituent properties in relation to that syntax?
- The "Component properties" section need not follow a definition list format. This could be converted to a bulleted list and maybe include just the salient points.
- Should a link to this guide also be added on the related property pages, the ones that are listed in the "Layer syntax" section?

wbamberg committed:
Fix security sidebar: remove "types of attacks" entry (#39778)

Fix security sidebar: remove types of attacks page

bsmth committed:
feat(fx): Sidebar for active Fx releases (#39761)

bsmth committed:
chore(learn): De-duplicate intro paras for Node and Django apps (#39767)

chrisdavidmills committed:
Editorial review: Update docs for Chrome 137 WebGPU additions (#39732)

* Update docs for Chrome 137 WebGPU additions

* Fix nit

* Fixes for hamishwillee review

OnkarRuikar committed:
Synchronize with BCD v6.0.20 (#39782)

Co-authored-by: OnkarRuikar <OnkarRuikar@users.noreply.github.com>

bsmth committed:
feat(fx): Improve Fx exp features page (#39765)

* feat: tidy up exp features tables

* feat: tidy up exp features tables

* feat: tidy up PR template

* Update files/en-us/mozilla/firefox/experimental_features/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

* Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

* Update files/en-us/mozilla/firefox/experimental_features/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

* chore(fx): fix dls in exp features page

* chore(fx): boolean to enum preference

* chore(fx): formatting fix

* Apply suggestions from code review

Co-authored-by: Vadim Makeev <hi@pepelsbey.dev>

---------

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>
Co-authored-by: Vadim Makeev <hi@pepelsbey.dev>

Josh-Cena committed:
Internalize role="timer" codepen demo (#39729)

Josh-Cena committed:
Remove redundant aria-live changes in timer example (#39787)

* Remove redundant role changes in timer example

* Change

chrisdavidmills committed:
Technical review: Add doc for the scope_extensions manifest member (#39649)

* Add doc for the scope_extensions manifest member

* Fix for dmurph review comment

dependabot[bot] committed:
Bump @mdn/yari from 4.10.1 to 4.11.0 (#39793)

chrisdavidmills committed:
Editorial review: SVG <use>: Omit URL fragment to reference external root (#39702)

* Add information about <use> element omitting URL fragment to load external root

* Update files/en-us/web/svg/reference/element/use/index.md

Co-authored-by: Vadim Makeev <hi@pepelsbey.dev>

* Update files/en-us/web/svg/reference/element/use/index.md

Co-authored-by: Vadim Makeev <hi@pepelsbey.dev>

* Update files/en-us/web/svg/reference/element/use/index.md

Co-authored-by: Vadim Makeev <hi@pepelsbey.dev>

---------

Co-authored-by: Vadim Makeev <hi@pepelsbey.dev>

Zwyx committed:
Add note about `beforeinstallprompt` event not fired if PWA already installed (#39792)

* Add note about `beforeinstallprompt` event not fired if PWA already installed

So users don't think using `getInstalledRelatedApps` is required for PWAs.

* Fix typo

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

* Add link to PWA installation criteria

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

* Remove unclear bit of text

---------

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

denmch committed:
Update index.md (#39772)

* Update index.md

The description for "Inclusion criteria" has a singular article with a plural noun: "an in-depth criteria." We could drop the article, but the sentence is still incomplete, as the subject is implied ("Provides…") rather than stated, as in the other examples (e.g., "This section covers," etc.). It seemed best to copy the introduction from "Inclusion criteria, which fixes both issues and keeps it consistent.

* Update index.md

Changed indent and removed clauser per wbamberg's comment. 

Under "Learn web development writing guidelines," I also corrected "this articles" to "this article" and removed the hyperlink on "Learn web development" in the first sentence.

I think it's confusing to include the link here because some folks will click "Learn web development" instead of "Learn web development writing guidelines" and end up in the wrong place.

* Update index.md

Added quotation marks for clarity.

chrisdavidmills committed:
Editorial review: Document canvas context lang attribute (#39642)

* Document canvas context lang attribute

* Add explanation of off-screen canvas inherit, and update demo to suit

* Add explanatory text to make purpose clear

* Update files/en-us/web/api/canvasrenderingcontext2d/lang/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/web/api/canvasrenderingcontext2d/lang/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/web/api/canvasrenderingcontext2d/lang/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Fixes for wbamberg review comments

* Update files/en-us/glossary/bcp_47_language_tag/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

---------

Co-authored-by: wbamberg <will@bootbonnet.ca>

Renegade334 committed:
Correct documentation around non-shared views to `Atomics.notify()` (#39799)

* javascript: correct global_objects/atomics/notify

* Update index.md

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

BeBoRE committed:
Fixed view-transition-class exampled (#39795)

helpimnotdrowning committed:
Fix uncapitalized letter S in start of paragraph (#39779)

Found while browsing the Elements list

ajuchacko committed:
Improves the explanation for the application-name metadata (#39584)

* Improves the explanation for the application-name metadata

* formatting with new line

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update files/en-us/web/html/reference/elements/meta/name/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

* Update files/en-us/web/html/reference/elements/meta/name/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

* Update files/en-us/web/html/reference/elements/meta/name/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update files/en-us/web/html/reference/elements/meta/name/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update files/en-us/web/html/reference/elements/meta/name/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update index.md

* Ran Prettier

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

chrisdavidmills committed:
Remove error from scope_extensions page (#39794)

jdatapple committed:
Fix Safari support for Add to Dock on macOS (#39797)

* Fix Safari support for Add to Dock on macOS

Safari has supported Add to Dock since macOS Sonoma.

* Update files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

* Update files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

---------

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>
